services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -314,72 +316,78 @@ async def manage_search_response(self, request: Request, response: StreamingResp
-        # update request body (better find the function that updates the body maybe?)c
-        request._body = json.dumps(content).encode("utf-8")  # pylint: disable=protected-access
-        return request  # pylint: disable=protected-access
-
-    async def manage_get_response(
-        self,
-        request: Request,
-        response: StreamingResponse,
-        ids: dict,
-    ) -> Response:
+        try:
+            user = self.request_ids["owner_id"]
+            content = await request.json()
+            if request.scope["path"] == "/collections":
+                content["id"] = f"{user}_{content['id']}"
+            if "items" in request.scope["path"]:
+                content = self.update_stac_item_publication(content, user)
+            elif request.scope["path"] == "/search" and "filter" in content:
+                qs_filter = content["filter"]
+                filters = parse_cql2_json(qs_filter)
+                user = self.find_owner_id(filters)
+                # May be duplicate?
+                if "collections" in content:
+                    content["collections"] = [f"{user}_{content['collections']}"]
+            # update request body (better find the function that updates the body maybe?)c
+            request._body = json.dumps(content).encode("utf-8")  # pylint: disable=protected-access
+            return request  # pylint: disable=protected-access
+        except KeyError as kerr_msg:
+            raise HTTPException(detail=f"Missing key in request body! {kerr_msg}", status_code=400) from kerr_msg
+
+    async def manage_get_response(self, request: Request, response: StreamingResponse) -> Response:
         """Remove the user name from obects and adapt all links.
 
         Args:
             request (Request): The client request.
             response (Response | StreamingResponse): The response from the rs-catalog.
-            ids (dict): a dictionnary containing owner_id, collection_id and
             item_id if they exist.
 
         Returns:
             Response: The response updated.
         """
-        user = ids["owner_id"]
+        user = self.request_ids["owner_id"]
         body = [chunk async for chunk in response.body_iterator]
         content = json.loads(b"".join(map(lambda x: x if isinstance(x, bytes) else x.encode(), body)).decode())
         if request.scope["path"] == "/":  # /catalog/owner_id
             return JSONResponse(content, status_code=response.status_code)
         if request.scope["path"] == "/collections":  # /catalog/owner_id/collections
             content["collections"] = filter_collections(content["collections"], user)
             content = self.remove_user_from_objects(content, user, "collections")
-            content = self.adapt_links(content, ids["owner_id"], ids["collection_id"], "collections")
+            content = self.adapt_links(
+                content,
+                self.request_ids["owner_id"],
```

Comment:
maybe we should not pass ids[] to parameters since we don't do it for manage_put_post_request ?

Reply:
Agree, can be removed if owner_id, collection_id from :297 are the same as the ones from self.request_ids.